.test-infra/jenkins/job_LoadTests_Java.groovy

```diff
@@ -17,123 +17,215 @@
+                        autoscalingAlgorithm: "NONE"
                 ]
         ],
         [
-                title        : 'GroupByKey load test Spark',
+                title        : 'Load test: reiterate 4 times 2MB values',
                 itClass      : 'org.apache.beam.sdk.loadtests.GroupByKeyLoadTest',
-                runner       : CommonTestProperties.Runner.SPARK,
-                sdk          : CommonTestProperties.SDK.JAVA,
+                runner       : CommonTestProperties.Runner.DATAFLOW,
                 jobProperties: [
-                        sparkMaster      : 'local[4]',
-                        publishToBigQuery: true,
-                        bigQueryDataset  : 'load_test_SMOKE',
-                        bigQueryTable    : 'spark_gbk',
-                        sourceOptions    : '{"numRecords":100000,"splitPointFrequencyRecords":1}',
-                        stepOptions      : '{"outputRecordsPerInputRecord":1,"preservesInputKeyDistribution":true}',
-                        fanout           : 10,
-                        iterations       : 1,
+                        project             : 'apache-beam-testing',
+                        appName             : 'load_tests_Java_Dataflow_Batch_GBK_7',
+                        tempLocation        : 'gs://temp-storage-for-perf-tests/loadtests',
+                        publishToBigQuery   : true,
+                        bigQueryDataset     : 'load_test',
+                        bigQueryTable       : 'java_dataflow_batch_GBK_7',
+                        sourceOptions       : """
+                                            {
+                                              "numRecords": 20000000,
+                                              "keySizeBytes": 10,
+                                              "valueSizeBytes": 90,
+                                              "numHotKeys": 10,
+                                              "hotKeyFraction": 1
+                                            }
+                                       """.trim().replaceAll("\\s", ""),
+                        fanout              : 1,
+                        iterations          : 4,
+                        maxNumWorkers       : 5,
+                        numWorkers          : 5,
+                        autoscalingAlgorithm: "NONE"
                 ]
         ]
 ]
 
+def loadTestJob = { scope ->
+  scope.description('Runs Java GBK load tests on Dataflow runner in batch mode')
+  commonJobProperties.setTopLevelMainJobProperties(scope, 'master', 240)
+
+  for (testConfiguration in loadTestConfigurations) {
+    loadTestsBuilder.loadTest(scope, testConfiguration.title, testConfiguration.runner, CommonTestProperties.SDK.JAVA, testConfiguration.jobProperties, testConfiguration.itClass, CommonTestProperties.TriggeringContext.PR)
+  }
+}
+
+CronJobBuilder.cronJob('beam_LoadTests_Java_GBK_Dataflow_Batch', 'H 12 * * *', this) {
+  loadTestJob(delegate)
+}
```

Comment:
Yes. It runs all GBK batch tests daily at 12 P.M. UTC time. Is that all right?

Reply:
As for SMOKE suites - we didn't schedule them yet to run. I guess since the real load tests are being commited we could run the smokes too. Should we? What should the trigger be (periodically/pre-commit/post-commit)?

Reply:
I'm not sure. Should we schedule them to run periodically? Or maybe just use them for debugging sometimes?

Reply:
So far we used it as you described (debugging, validating PRs on demand). IMO let's keep it this way - It's not a paramount or easy to break piece of code. If in the future, if we'll have problems with that, it's always easy to change the triggers for smokes to be pre/post commit (1 line change?)